instagram/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+// import Proptypes from 'prop-types';
+
+import './SearchBar.scss';
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchQuery: '',
+    }
+  }
+
+  handleInputChange = (event) => {
+    console.log('EVENT', event.target.value)
+    const searchQuery = event.target.value;
+
+    this.setState({
+      searchQuery: searchQuery,
+    });
+
+    if (searchQuery) return;
+
+    // When there's no search value or the clear button is pressed in the input field
+    this.props.handleSubmit(searchQuery);
```

Comment:
Another thing to keep in mind is that you can have an input without a form. Your searchQuery string can be held in state without a submit. That's what the changehandler is doing